l1-contracts/contracts/state-transition/chain-deps/facets/Admin.sol

```diff
@@ -2,20 +2,21 @@
 
 pragma solidity 0.8.20;
 
-import {IAdmin} from "../interfaces/IAdmin.sol";
-import {Diamond} from "../libraries/Diamond.sol";
-import {MAX_GAS_PER_TRANSACTION} from "../Config.sol";
-import {FeeParams} from "../Storage.sol";
-import {Base} from "./Base.sol";
+import {IAdmin} from "../../chain-interfaces/IAdmin.sol";
+import {Diamond} from "../../libraries/Diamond.sol";
+import {MAX_GAS_PER_TRANSACTION} from "../../../common/Config.sol";
+import {FeeParams} from "../ZkSyncStateTransitionStorage.sol";
+import {ZkSyncStateTransitionBase} from "./Base.sol";
+import {IStateTransitionManager} from "../../IStateTransitionManager.sol";
 
 // While formally the following import is not used, it is needed to inherit documentation from it
-import {IBase} from "../interfaces/IBase.sol";
+import {IZkSyncStateTransitionBase} from "../../chain-interfaces/IBase.sol";
 
 /// @title Admin Contract controls access rights for contract management.
 /// @author Matter Labs
 /// @custom:security-contact security@matterlabs.dev
-contract AdminFacet is Base, IAdmin {
-    /// @inheritdoc IBase
+contract AdminFacet is ZkSyncStateTransitionBase, IAdmin {
```

Comment:
Correct me if I am wrong. Currently it is hard to enforce custom consensus for STC. AFAIU, the current way for the chain to set up custom consensus rules are setting some kind of `ValidatorTimelock` (a wrapper over real validator) as a validator. But the problem that I see is that `validator` can be changed to anything instantly by the governor or admin. So it is not set in stone and I would expect that someone would want it to be enforced and be immutable as it may crush tokenomics

Reply:
@vladbochok maybe like this:

- ValidatorTimelock = consensus
- Governor/Admin = rules for changing it? 

They can just as well set some limited Governor that can only execute upgrades requested by the main STM

Reply:
Yes, but it is a bit complex tbh. Should we clean this up?

Reply:
I am not totally sure how it could be cleaned up. 

STCs must be upgradable => some governance is needed. Also we give the freedom for each STC to have their own governance. So it is the job of each STC to decide how their governance should look like. In theory, if upgrdability was not a concern, they could've set governance to zero (or some random address that def wont have anything deployed there).

Reply:
True. Can they not specify in their governor contract what they can change?

Reply:
I agree with Stas. But on another matter, I think actually we should make the Validator timelock enforced initially. Wdyt? (plural you)

Reply:
Imagine there is a chain X, it wants to have a governance for changing fee params, upgrading chains, unfreezing chain. But Governance should not be able to remove `consensus`. So this chain should reuse our Governance contract with changes to disallow changing validator?

I see cleaner solution as 
1) removing `validators` completely from the STC
2) Adding `consensus` address to the STC storage
3) `consensus` would be a part of `initCalldata` when STM creates new STC

Reply:
I want to avoid inexplicit checks on the Governance that should know implementation of the STC, then we can probably generalise `Governance.sol` for most of needs without separate versions and hacks

Reply:
Agree with @kelemeno that we should enforce our security mechanisms for all hyperchains

Reply:
> that we should enforce our security mechanisms for all hyperchains

Makes sense, especially to avoid situations when someone could circumvent the situation by transferring funds through another chain

Reply:
But sometimes even the consensus mechanism is changed, (the merge). So governance can sometimes change that as well. But I'm okay with removing `validators`  in favor of `consensus`, but I would keep the privileges of governance the same.